### PR TITLE
Constants

### PR DESCRIPTION
**Pull request by jan-dubsky**
To be able to parse some `varnishlog` output, we need to have constants we can look for in the log. And that is exactly what this diff adds. The list of constants is by far not complete, but it's a good start. More constants might be added over time.

### CODE REVIEW

File: entry.go [hunk 1]
```diff
@@ -1,11 +1,23 @@
 package vslparser
 
+// VXID is varnish transaction ID.
+//
+// In Varnish code, VXIDs are represented as uint32_t type and the value wraps
+// at 1<<30, so Go uint32 us absolutely sufficient to hold any possible value of
+// VXID.
+//
+// Note: Zero value (vxid == 0) is not reported in any varnishlog mode with an
+// exception or 'raw' mode. Consequently, zero value of VXID can be in some
```
Comment (Hrubon) [inline entry.go:10]: or -> of?

File: keys.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	// KindRequest is a Kind string identifying Varnish Request object.
+	KindRequest = "Request"
+	// KindBeReq is a Kind string identifying Varnish BeReq object.
+	KindBeReq = "BeReq"
+	// KindSession is a Kind string identifying Varnish Session objects.
+	KindSession = "Session"
+)
+
+const (
+	// TagBegin is tag key informing about begin of tags section (start of
```
Comment (Hrubon) [inline keys.go]: begin -> beginning

File: keys.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	// VXID).
+	TagBegin = "Begin"
+	// TagEnd is tag key informing about end of tags section (end of VXID).
```
Comment (Hrubon) [inline keys.go:16]: end -> ending
But I am not sure. :) Check English rules.

File: keys.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	// TagReqURL is a tag key identifying request URL.
+	TagReqURL = "ReqURL"
+	// TagReqProtocol is a tag key identifying HTTP protocol version.
+	TagReqProtocol = "ReqProtocol"
+	// TagReqMethod is a tag key identifying HTTP request method verb.
+	TagReqMethod = "ReqMethod"
+	// TagRespStatus is a tag key identifying HTTP response status code.
+	TagRespStatus = "RespStatus"
+
+	// TagReqHeader is a tag informing that Request header was set.
```
Comment (Hrubon) [inline keys.go]: informing -> indicating?
The same for the tags below.

File: tag_set.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		lookup[tag.Key] = append(lookup[tag.Key], tag)
+	}
+
+	return TagSet{
+		lookup: lookup,
+		list:   tags,
+	}
+}
+
+func (t *TagSet) FirstWithKey(key string) (Tag, bool) { return t.NthWithKey(1, key) }
```
Comment (Hrubon) [inline tag_set.go]: Just a nitpick: I personally like ordering the functions in the order which one calls another. So maybe consider placing `FirstWithKey` below `NthWithKey`? :)

File: tag_set.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+func (t *TagSet) NthWithKey(n int, key string) (Tag, bool) {
```
Comment (Hrubon) [inline tag_set.go]: Another nitpick: I know that you probably don't do that anywhere in the code, but what if somebody will call this function with `n=0` or even `n < 0`? After all, this is public package, and this method is part of its public API, so such corner-cases probably should be taken into account, especially when the corner-case leads to `panic`.

File: tags.go [hunk 7]
```diff
@@ -1,47 +1,40 @@
...
-	lookup map[string][]Tag
-	list   []Tag
+// Tags is an array of VSL tags with added functionality.
+//
+// Contrary to TagSet, Tags are much cheaper to create as it doesn't perform any
+// allocations, nor any smart handling. On the other hand searches in plain Tags
+// take linear time.
+type Tags []Tag
+
+// FirstWithKey finds first tag with key key in t. This method takes O(n).
```
Comment (Hrubon) [inline tags.go]: key key -> key

**Review by Hrubon — CHANGES_REQUESTED** [review]
Mostly English-related notes in comments.

Please address the comment regarding `NthWithKey` - this is the reason why I've requested changes.

File: tags.go (hunk 7 above)
Comment (jan-dubsky) [reply tags.go]: Not really. The key is called key :trollface: But let's change it to `k` ;)

File: tag_set.go (hunk 6 above)
Comment (jan-dubsky) [reply tag_set.go]: OK, I have redefined the API not to allow any invalid usage.

File: tag_set.go (hunk 5 above)
Comment (jan-dubsky) [reply tag_set.go]: It makes more sense to have them ordered `First`, `Nth`, `Last` than to interleave those. But I agree it lacks the C-like vibe.

**Review by Hrubon — APPROVED** [review]
Thanks, looks good now and LGTM.

**Pull request merged**